app/javascript/packs/Profile.jsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { isEmpty } from 'lodash';
+
+export default class Profile extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      posterBasePath: "https://image.tmdb.org/t/p/w154",
+      movies: [],
+    }
+    this.removeMovie = this.removeMovie.bind(this);
+  }
+
+  componentWillMount() {
+    const self = this;
+    const { store } = this.context;
+    const user = store.getState().user[0];
+
+    axios({
+      method: 'GET', 
+      url: '/users/'+user.id+'/movies',
+      data: {
+      },
+    }).then(function(response) {
+      self.setState({ movies: response.data.movies });
+      self.context.store.dispatch({
+        type: 'USER_MOVIES',
+        data: response.data.movies,
+      });
+    }).catch(function(error) {
+      console.log("Movie search blew up!", error);
+    });
+  }
+
+  componentDidMount() {
+    const { store } = this.context;
+    this.unsubscribe = store.subscribe(() =>
+      this.forceUpdate()
+    );
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  removeMovie(e) {
+    e.preventDefault();
+    const self = this;
+    const userID = self.context.store.getState().user[0].id;
+    const parentDiv = e.target.parentElement.parentElement.parentElement;
+
+    e.target.innerText = 'Movie Removed';
+    e.target.disabled = true;
+    setTimeout(function() { parentDiv.remove() }.bind(this), 3000);
+
+    const movie = this.state.movies.find(function findMovie(movie) {
+      if (movie.title === e.target.parentElement.children[0].innerText) {
+        return movie;
+      }
+    });
+
+    axios({
+      method: 'DELETE',
+      url: '/users/'+userID+'/movies/'+movie.id,
+      data: {
+        title: movie.title,
+        overview: movie.overview,
+        release_date: movie.release_date,
+        poster_path: this.state.posterBasePath+movie.poster_path,
+        user_id: userID,
+      },
+      headers: {
+        'X-CSRF-Token': document.querySelector("meta[name=csrf-token]").content,
+        'Content-Type': 'application/json',
+      }
+    }).then(function(response) {
+    }).catch(function(error) {
+      self.setState({ errors: error.response.data })
+    });
+  }
+
+  render() {
+    const movies = this.state.movies;
+    const user = this.context.store.getState().user[0];
+
+    return (
+      <div className="container target">
```

Comment:
a common pattern we like to use is having smart and dumb components

https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0